docs/conditional-gatherer/README.md

```diff
@@ -1,185 +1,137 @@
 
-1. Run the service following the instructions here
-   https://github.com/RedHatInsights/insights-operator-gathering-conditions-service
-2. Set `conditionalGathererEndpoint` in `config/local.yaml` to `http://localhost:8081/api/gathering/gathering_rules`
-3. Enjoy your conditional rules from the local service
+| Field                         | Type   | Description                                                  | Required |
+|-------------------------------|--------|--------------------------------------------------------------|----------|
+| `namespace`                   | string | The namespace for log gathering.                             | Required |
+| `namespaces`                  | array  | An array of namespaces (future replacement for `namespace`). | Optional |
+| `label_selector`              | string | Label selector for filtering containers.                     | Optional |
+| `field_selector`              | string | Field selector for filtering containers.                     | Optional |
+| `container_name_regex_filter` | string | Regex filter for container names.                            | Optional |
+| `pod_name_regex_filter`       | string | Regex filter for pod names.                                  | Optional |
+| `max_namespace_containers`    | int    | Maximum number of containers to include.                     | Optional |
 
-## Using a Mock Server
 
-1. Start a mock server:
-```bash
-git clone https://github.com/RedHatInsights/insights-operator-gathering-conditions.git
-cd insights-operator-gathering-conditions/
-./build.sh
-python3 -m http.server --directory build/
-```
-
-2. Set `conditionalGathererEndpoint` in `config/local.yaml` to `http://localhost:8000/rules.json`
-3. Enjoy your conditional rules from the mock service
-
-## Using Stage Endpoint
-
-0. Be connected to Red Hat network or configure a proxy for stage version of console.redhat.com
-1. Set up the stage endpoint in `config/local.yaml`
-2. Configure authentication through support secret
-```bash
-echo '{
-  "apiVersion": "v1",
-  "kind": "Secret",
-  "metadata": {
-    "namespace": "openshift-config",
-    "name": "support"
-  },
-  "type": "Opaque",
-  "data": {
-    "username": "'(echo $STAGE_USERNAME | base64 --wrap=0)'",
-    "password": "'(echo $STAGE_PASSWORD | base64 --wrap=0)'"
-  }
-}' | oc apply -f -
-```
+#### Log filter
+The `log_filter` structure enables users to filter log messages. It includes the following fields:
 
-3. Enjoy your conditional rules from the stage endpoint
+| Field                   | Type   | Description                                           | Required |
+|-------------------------|--------|-------------------------------------------------------|----------|
+| `messages_to_search`    | array  | Array of messages to search for.                      | Required |
+| `is_regex_search`       | bool   | Boolean flag for regex search.                        | Optional |
```

Comment:
I would prefer to have `messages` and `messages_regex` parameters over having an optional parameter whose sole purpose is to modify the behavior of another optional parameter.  I would be fine with optional parameters modifying the behavior of mandatory parameters, but this is not the case.

Reply:
Not sure if part of the documentation existing feature task, however, I'm taking these notes for future improvements, since it needs some effort to change.

Reply:
I don't understand why you are so opposed to iterating over the rule specification. I don't care about the code that implements the gathering rules at this point. Users should not be expected to read the Insights Operator source code to be able to write gathering rules.

Reply:
OK. Let's do it... We can split it into `messages` and `messages_regex` for the conditions, but basically, the code would still behavior the same, it would be only an interface to interact with the code that we already have, so we can mitigate the effort to change it.

Reply:
I repeat: I don't care about the code in this PR. The code doesn't do what we need. If nothing else, it does not provide the logs in the format that was requested. That alone might mean that you will need to reconsider the whole gathering approach.

Reply:
I know that you don't care about the code, but I do. 😆 

I see no reason why `message` + `message_regex` instead `message_to_search` + `is_regex_seach` would not "provide the logs in the format that was requested". But, as I said, looking a the code it is possible to adjust to support both parameters `message` + `message_regex` instead of the current implementation, and from the implementation perspective it would not cause any impact, since under the hood it would still using the same implementation.

So, what I'm saying is: yes, let's change to `message` and `message_regex`, it doesn't seem to be such a difficult implementation as I was thinking.

Reply:
You are providing feedback on how my proposal affects the current implementation. I am expecting feedback on the API that the gathering rules comprise - what is more intuitive/clear/powerful?

Having said that, we could also say that the messages have to be regexes and handle non-regex patters on the gathering rule creation side.